app/services/parent_profile_associator.rb

```diff
@@ -14,18 +14,46 @@ def run!
     private
 
     def find_parent(profile)
-      parents = Profile.where(ref_id: profile.ref_id,
-                              benchmark_id: profile.benchmark_id,
-                              account_id: nil)
-      raise not_found(profile) unless parents.one?
+      logger.info("Finding parents for profile #{profile.id}")
+      parents = find_in_same_benchmark(profile)
+      parents = find_in_other_benchmarks(profile) if parents.blank?
 
-      parents.first
+      raise not_found(profile) unless parents.any?
+
+      parent = find_most_recent_parent(parents)
+      logger.info("Found parent #{parent.id} for profile #{profile.id}")
+      parent
+    end
+
+    def find_in_same_benchmark(profile)
+      Profile.where(ref_id: profile.ref_id,
+                    benchmark_id: profile.benchmark_id,
+                    account_id: nil)
+    end
+
+    # Profiles in the phony-benchmark will have to be assigned to
+    # a real benchmark if it can be found.
+    def find_in_other_benchmarks(profile)
+      Profile.where(name: profile.name,
```

Comment:
I'm matching both the name and the ref_id here as that seems a more accurate way of finding the parent profile outside of its own benchmark

Reply:
We could (maybe should) add an index on name, since we seem to need to search by name like this periodically. After this uniqueness stuff is solved, I plan on going through our schema and adding indexes on any searched fields